invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -201,71 +197,68 @@ class DataCite43Schema(Schema):
         fields.Nested(FundingSchema43), attribute='metadata.funding')
     schemaVersion = fields.Constant("http://datacite.org/schema/kernel-4")
 
-    def _read_resource_type(self, id_):
-        """Retrieve resource type record using service."""
-        return vocabulary_service.read(
-            ('resource_types', id_),
-            system_identity
-        )
-
-    def _read_title_type(self, id_):
-        """Retrieve title type record using service."""
-        return vocabulary_service.read(
-            ('title_types', id_),
-            system_identity
+    def _map_type(self, vocabulary, fields, id_):
```

Comment:
This pattern with the props is used also before with the `Contributor.role`, `locations` below... maybe we can extract it a bit more? MIght not be worth if the code becomes too messy since there is only a few appearances...

Reply:
For now whe have the map_type that is used widely and also the one used for `contributors.role`, I would keep it as it for now, and refactor it, if needed when we finish subjects/affiliations